### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by thegwan** [smartcorelib/smartcore#260] [Q&A] [answered]
Are there any performance expectations for the algorithms implemented in this library? Should I expect them to be comparable to or faster than equivalent scikit-learn implementations?

My anecdotal tests show that they are quite a bit slower in both training and inference (I've been playing with decision tree in particular), but would like to confirm if this is expected.

**Comment by Mec-iS** [comment, marked as answer]
Hi,

Thanks for using the library. Indeed pure Rust implementations are still slower, 
Python implementations (or other relying on C++ bindings) relies on lower-level implementations that had decades to be optimised by hundreds of developers, this project tries the long shot of starting things from scratch.
We definitely need some profiling so it would be great if you can help with that.

Best,
